### PR TITLE
pkg: add error handling

### PR DESCRIPTION
**Pull request by hwdef**
Signed-off-by: hwdef <hwdef97@gmail.com>

### Ⅰ. Describe what this PR did
add error handling

### Ⅱ. Does this pull request fix one issue?

ref: #952

### CODE REVIEW

File: pkg/digest/digest.go [hunk 1]
```diff
@@ -34,7 +35,9 @@ func Sha256(value string) string {
 func Sha1(contents []string) string {
 	h := sha1.New()
 	for _, content := range contents {
-		io.WriteString(h, content)
+		if _, err := io.WriteString(h, content); err != nil {
+			fmt.Println(err)
```
Comment (allencloud) [inline pkg/digest/digest.go]: `fmt.Println(err)` would send error message to the stdout, I am wondering if we should send it to stderr.

In addition, I wish to see the principle that where indeed we should handle the returned error?

File: pkg/fileutils/fileutils.go [hunk 2]
```diff
@@ -60,7 +60,9 @@ func DeleteFile(filePath string) error {
 func DeleteFiles(filePaths ...string) {
 	if len(filePaths) > 0 {
 		for _, f := range filePaths {
-			DeleteFile(f)
+			if err := DeleteFile(f); err != nil {
+				fmt.Println(err)
```
Comment (allencloud) [inline pkg/fileutils/fileutils.go]: same issue.

File: pkg/digest/digest.go (hunk 1 above)
Comment (hwdef) [reply pkg/digest/digest.go]: I think the error in this function that should not return.
becouse this func be called in append() func.
can only output.

Comment (allencloud) [reply pkg/digest/digest.go]: I do not mean `return err`, but stream error message into stderr rather than stdout.

Comment (hwdef) [reply pkg/digest/digest.go]: I see,I fixed it.

File: pkg/digest/digest.go [hunk 3]
```diff
@@ -34,7 +36,9 @@ func Sha256(value string) string {
 func Sha1(contents []string) string {
 	h := sha1.New()
 	for _, content := range contents {
-		io.WriteString(h, content)
+		if _, err := io.WriteString(h, content); err != nil {
+			fmt.Fprintf(os.Stderr, "fail to write string to io.Write %v", err)
```
Comment (allencloud) [inline pkg/digest/digest.go]: failed to write string to io.Write: %v

please use `failed` and `:`

Comment (hwdef) [reply pkg/digest/digest.go]: ok, I add them.

File: pkg/fileutils/fileutils.go [hunk 4]
```diff
@@ -60,7 +60,9 @@ func DeleteFile(filePath string) error {
 func DeleteFiles(filePaths ...string) {
 	if len(filePaths) > 0 {
 		for _, f := range filePaths {
-			DeleteFile(f)
+			if err := DeleteFile(f); err != nil {
+				fmt.Fprintf(os.Stderr, "fail to delete file: %v", err)
```
Comment (yeya24) [inline pkg/fileutils/fileutils.go]: Please keep compatibility here. s/fail/failed/g .  Also, we need to log the failed file path in the error message.

File: pkg/digest/digest.go [hunk 5]
```diff
@@ -34,7 +36,9 @@ func Sha256(value string) string {
 func Sha1(contents []string) string {
 	h := sha1.New()
 	for _, content := range contents {
-		io.WriteString(h, content)
+		if _, err := io.WriteString(h, content); err != nil {
+			fmt.Fprintf(os.Stderr, "failed to write string to io.Write: %v", err)
```
Comment (yeya24) [inline pkg/digest/digest.go]: @hwdef  I don' t think we need an error handling here.
If you really want to add it here, it is better to change the error description.
Maybe failed to calculate sha-256 hash?

Comment (hwdef) [reply pkg/digest/digest.go]: I think you are right.
I will close this PR.

**Pull request closed without merging**